File: 2129-number-of-pairs-of-interchangeable-rectangles/2129-number-of-pairs-of-interchangeable-rectangles.java
class Solution {
    public long interchangeableRectangles(int[][] rectangles) {
      Map<Double,Long> map=new HashMap<>();
      long count=0;
      for(int[] num:rectangles){
       double a=num[0];
       double b=num[1];
       double division =(double) a/b;
       if(map.containsKey(division)){
        count+=map.put(division,map.get(division)+1);
       }else{
       map.put(division,1L);
       }
      }
      return count;

    }

}

File: 2129-number-of-pairs-of-interchangeable-rectangles/2129-number-of-pairs-of-interchangeable-rectangles.cpp
class Solution {
public:
    long long interchangeableRectangles(vector<vector<int>>& rectangles) {
        long long pc = 0;
        map<double, int> m;
        int n = rectangles.size();
        for(int i = 0; i < n; i++) {
            double v = (double)rectangles[i][0] / rectangles[i][1];
            pc = pc + m[v];
            m[v] = m[v] + 1;
        }
        return pc;
    }
};